src/docfx/template/TemplateEngine.cs

```diff
@@ -37,21 +35,22 @@ internal class TemplateEngine : IDisposable
             _config = config;
             _output = output;
 
-            _templateDir = config.Template.Type switch
+            var template = config.Template;
+            if (template.Type == PackageType.None)
             {
-                PackageType.None => Path.Combine(buildOptions.DocsetPath, "_themes"),
-                _ => packageResolver.ResolvePackage(config.Template, PackageFetchOptions.DepthOne),
-            };
+                template = new PackagePath(Path.Combine(buildOptions.DocsetPath, "_themes"));
+            }
+
+            _package = packageResolver.ResolvePackage2(template, PackageFetchOptions.DepthOne);
 
-            _contentTemplateDir = Path.Combine(_templateDir, "ContentTemplate");
-            _schemaDir = Path.Combine(_contentTemplateDir, "schemas");
+            _templateDefinition = new Lazy<TemplateDefinition>(() =>
+                _package.TryReadYamlOrJson<TemplateDefinition>(errors, "template") ?? new TemplateDefinition());
 
-            _templateDefinition = PathUtility.LoadYamlOrJson<TemplateDefinition>(errors, _templateDir, "template") ?? new TemplateDefinition();
+            _global = _package.TryReadYamlOrJson<JObject>(errors, "ContentTemplate/token") ?? new JObject();
 
-            _global = LoadGlobalTokens();
-            _liquid = new LiquidTemplate(_templateDir, _global);
-            _js = new ThreadLocal<JavaScriptEngine>(() => JavaScriptEngine.Create(_contentTemplateDir, _global));
-            _mustacheTemplate = new MustacheTemplate(_contentTemplateDir, _global, jsonSchemaTransformer);
+            _liquid = new LiquidTemplate(_package, _global);
+            _js = new ThreadLocal<JavaScriptEngine>(() => JavaScriptEngine.Create(_package, _global));
```

Comment:
Consider passing the basePath `contentTemplate` to JS engine when creating it, so no need to prepend it when run the scripts